lib/session.ex

```diff
@@ -117,40 +97,31 @@ defmodule Janus.Session do
     GenServer.call(session, {:execute_message, message, timeout})
   end
 
-  @doc """
-  Replaces current connection with a new one.
-
-  ## Arguments
-
-  * `session` - a PID of `Janus.Session` process.
-  * `connection` - a PID of `Janus.Connection` process.
-
-  ## Return values
-
-  Returns an `:ok` atom.
-  """
-  @spec update_connection(Janus.Session.t(), connection_t()) :: :ok
-  def update_connection(session, connection) do
-    GenServer.cast(session, {:new_connection, connection})
-  end
-
   # callbacks
 
   @impl true
-  def init({session_id, connection}) do
-    state = %{
-      session_id: session_id,
-      connection: connection
-    }
+  def init({connection, timeout}) do
+    case Connection.call(connection, %{janus: :create}, timeout) do
+      {:ok, %{"id" => session_id}} ->
+        state = %{session_id: session_id, connection: connection}
+
+        interval = Janus.Connection.get_transport_module(connection).keepalive_interval()
 
-    keep_alive = Janus.Connection.get_transport_module(connection).keepalive_timeout()
+        state =
+          case interval do
+            nil ->
+              state
+
+            interval ->
+              Process.send_after(self(), :keep_alive, interval)
+              Map.put(state, :keepalive_interval, interval)
+          end
```

Comment:
It won't hurt to set `:keepalive_interval` even if it's `nil`, so I think we can make it more readable:
```suggestion
        state = Map.put(state, :keepalive_interval, interval)

        unless is_nil(interval) do
          Process.send_after(self(), :keep_alive, interval)
        end
```